h/services/oauth_provider.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import datetime
+
+from oauthlib.oauth2 import (
+    AuthorizationCodeGrant,
+    AuthorizationEndpoint,
+    BearerToken,
+    RefreshTokenGrant,
+    TokenEndpoint,
+)
+
+from h.oauth import InvalidRefreshTokenError, JWTAuthorizationGrant
+from h.security import token_urlsafe
+
+TOKEN_TTL = datetime.timedelta(hours=1).total_seconds()
+
+
+class OAuthProviderService(AuthorizationEndpoint, TokenEndpoint):
```

Comment:
I don't know about you, but I find I often miss parent classes and mixins when first scanning code. It might be a good idea to add a docstring here explaining what OAuthProviderService is for, and what AuthorizationEndpoint and TokenEndpoint to (or at least point to their docs).